### DISCUSSION TITLE
Unrecoverable crash

### DISCUSSION THREAD

**Discussion by 1800joe** [noop-inc/documentation#66] [Bugs] [answered]
Occasionally the app becomes unresponsive with loading indicators on all dashboard panels.

<img width="1816" alt="Screenshot 2024-01-12 at 10 20 40 AM" src="https://github.com/noop-inc/documentation/assets/3111670/47376fb7-6732-4d63-8f86-083c13421721">

Seems to happen at unexpected times, maybe more reliably when two applications are deploying at the same time

Edit:
Sometimes on restart the crash state persists:

<img width="1392" alt="Screenshot 2024-01-12 at 10 24 00 AM" src="https://github.com/noop-inc/documentation/assets/3111670/c3787c7c-9cbb-41e4-a8d6-a259668dee68">

Edit:
Appears to happen if there is any file in the .noop directory other than the blueprint.yaml

Edit:
Appears to be something specific to Phoenix Template

**Comment by arbitrarytech** [comment]
@1800joe there is definitely some specific issues with the Phoenix Template as it spikes up to ~5GB of memory during the Build an then crashes. These memory issues right now threaten overall Workshop VM stability and we are working on fixes that will manage memory better.

**Reply by arbitrarytech** [reply, marked as answer]
Improvements have been made in v0.3.1 to better manage system memory inside the WorkshopVM. I've tested the Phoenix template further, and while it always spikes up to consume all VM memory during the Build, it should only kill the Build and no longer cause Workshop to become unresponsive. Let me know if you experience any further issues, marking answered for now.
